### ISSUE TITLE
Add a Reporting System that calculates the total Uptime percentage of my application

### ISSUE THREAD

**Issue by rha-wtag** [TwiN/gatus#855]
### Describe the feature request

My clients require a report of the amount of time my application faced a downtime. 
For example: 100% Uptime in last 1 week
95% uptime in last 365 days.

**Comment by johanjk** [comment]
The simplest workaround would be to:
 - use gatus with sql backend.
 - Realize there is an endpoint for [uptime badges](https://github.com/TwiN/gatus/tree/master?tab=readme-ov-file#uptime), doing nearly what you require.
 - Take a look at the source code for the uptime badge generation: https://github.com/TwiN/gatus/blob/12b04802ef1f81a16dd1a19b7d5e1373a01264bc/storage/store/sql/sql.go#L782-L787 
 - Write your own query, interfacing with the gatus db.

**Comment by melroy89** [comment]
See also: https://github.com/TwiN/gatus/issues/1646

People really want an uptime percentage.

**Comment by CarmJos** [comment]
> See also: [#1646](https://github.com/TwiN/gatus/issues/1646)
> 
> People really want an uptime percentage.

Yeah. I couldn't wait any longer and decided to implement it myself.

 I’ve submitted a PR (#1651 )and hope it gets merged soon. You can now pull my version for testing via [GitHub Packages](https://github.com/CarmJos/gatus/pkgs/container/gatus).

```shell
docker run -p 8080:8080 -v /srv/status:/config --name gatus ghcr.io/carmjos/gatus:0.0.8
```
